lib/queue.js

```diff
@@ -92,55 +92,64 @@ var Queue = function Queue(name, url, opts){
     throw Error('Options must be a valid object');
   }
 
-  var redisOpts = opts.redis || {};
+  this.name = name;
+  this.token = uuid();
 
-  _.defaults(redisOpts, {
+  opts.redis = opts.redis || {};
+
```

Comment:
The problem with this approach is that we are modifying the passed opts object which could create side effects on  the user's code. However I am seeing that we are already creating side effects since redisOpts is not a copy of opts.redis but a reference. I think it would be wise to clone the options before modifying them.